python/pyarrow/_parquet.pyx

```diff
@@ -1221,8 +1222,8 @@ cdef class ParquetReader(_Weakrefable):
                 string_to_timeunit(coerce_int96_timestamp_unit))
 
         self.source = source
+        self.nf = get_reader(source, use_memory_map, &rd_handle)
```

Comment:
Just for the record, is it necessary to store the `NativeFile` or could you also simply store the `shared_ptr[CRandomAccessFile]`? Not that it makes much of a difference, I suppose.

Reply:
I tried it initially, but had pickling tests failing. Second time around seems to work: https://github.com/apache/arrow/pull/13821/commits/a3eff73872bfb27a690033ab237cd7184158e263